embed/ContainerTrait.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\mongodb\embed;
+
+use yii\base\InvalidConfigException;
+use yii\base\InvalidParamException;
+use Yii;
+
+/**
+ * ContainerTrait can be used to satisfy [[ContainerInterface]].
+ *
+ * For each embed entity a mapping declaration should be provided.
+ * In order to do so you need to declare method, which name is prefixed with 'embed', which
+ * should return the [[Mapping]] instance. You may use [[hasEmbed()]] and [[hasEmbedList()]] for this.
+ *
+ * Per each of source field or property a new virtual property will declared, which name will be composed
+ * by removing 'embed' prefix from the declaration method name.
+ *
+ * Note: watch for the naming collisions: if you have a source property named 'profile' the mapping declaration
+ * for it should have different name, like 'profileModel'.
+ *
+ * Example:
+ *
+ * ~~~php
+ * use yii\base\Model;
+ * use yii\mongodb\embed\ContainerInterface;
+ * use yii\mongodb\embed\ContainerTrait;
+ *
+ * class User extends Model implements ContainerInterface
+ * {
+ *     use ContainerTrait;
+ *
+ *     public $profileData = [];
+ *     public $commentsData = [];
+ *
+ *     public function embedProfile()
+ *     {
+ *         return $this->hasEmbed('profileData', 'Profile');
```

Comment:
@klimov-paul I mean... aside from the obvious

class Profile extends Model implements ContainerInterface {
	use ContainerTrait;

	public $firstName;
	public $lastName;
}

should there be anything else?